Mathlib/CategoryTheory/Bicategory/NaturalTransformation/Strong.lean

```diff
@@ -0,0 +1,202 @@
+  rfl
+
+instance : Inhabited (StrongOplaxNatTrans F F) :=
+  ⟨id F⟩
+
+variable {F} {G H : OplaxFunctor B C} (η : StrongOplaxNatTrans F G) (θ : StrongOplaxNatTrans G H)
+
+section
+
+variable {a b c : B} {a' : C}
+
+@[reassoc (attr := simp)]
+theorem whiskerLeft_naturality_naturality (f : a' ⟶ G.obj a) {g h : a ⟶ b} (β : g ⟶ h) :
+    f ◁ G.map₂ β ▷ θ.app b ≫ f ◁ (θ.naturality h).hom =
+      f ◁ (θ.naturality g).hom ≫ f ◁ θ.app a ◁ H.map₂ β := by
+  apply θ.toOplax.whiskerLeft_naturality_naturality
+
+@[reassoc (attr := simp)]
+theorem whiskerRight_naturality_naturality {f g : a ⟶ b} (β : f ⟶ g) (h : G.obj b ⟶ a') :
+    F.map₂ β ▷ η.app b ▷ h ≫ (η.naturality g).hom ▷ h =
+      (η.naturality f).hom ▷ h ≫ (α_ _ _ _).hom ≫ η.app a ◁ G.map₂ β ▷ h ≫ (α_ _ _ _).inv := by
+  apply η.toOplax.whiskerRight_naturality_naturality
+
+@[reassoc (attr := simp)]
+theorem whiskerLeft_naturality_comp (f : a' ⟶ G.obj a) (g : a ⟶ b) (h : b ⟶ c) :
+    f ◁ (θ.naturality (g ≫ h)).hom ≫ f ◁ θ.app a ◁ H.mapComp g h =
+      f ◁ G.mapComp g h ▷ θ.app c ≫
+        f ◁ (α_ _ _ _).hom ≫
+          f ◁ G.map g ◁ (θ.naturality h).hom ≫
+            f ◁ (α_ _ _ _).inv ≫ f ◁ (θ.naturality g).hom ▷ H.map h ≫ f ◁ (α_ _ _ _).hom := by
+  apply θ.toOplax.whiskerLeft_naturality_comp
+
+@[reassoc (attr := simp)]
+theorem whiskerRight_naturality_comp (f : a ⟶ b) (g : b ⟶ c) (h : G.obj c ⟶ a') :
+    (η.naturality (f ≫ g)).hom ▷ h ≫ (α_ _ _ _).hom ≫ η.app a ◁ G.mapComp f g ▷ h =
+      F.mapComp f g ▷ η.app c ▷ h ≫
+        (α_ _ _ _).hom ▷ h ≫
+          (α_ _ _ _).hom ≫
+            F.map f ◁ (η.naturality g).hom ▷ h ≫
+              (α_ _ _ _).inv ≫
+                (α_ _ _ _).inv ▷ h ≫
+                 (η.naturality f).hom ▷ G.map g ▷ h ≫ (α_ _ _ _).hom ▷ h ≫ (α_ _ _ _).hom := by
+  apply η.toOplax.whiskerRight_naturality_comp
+
+@[reassoc (attr := simp)]
+theorem whiskerLeft_naturality_id (f : a' ⟶ G.obj a) :
+    f ◁ (θ.naturality (𝟙 a)).hom ≫ f ◁ θ.app a ◁ H.mapId a =
+      f ◁ G.mapId a ▷ θ.app a ≫ f ◁ (λ_ (θ.app a)).hom ≫ f ◁ (ρ_ (θ.app a)).inv := by
+  apply θ.toOplax.whiskerLeft_naturality_id
+
+@[reassoc (attr := simp)]
+theorem whiskerRight_naturality_id (f : G.obj a ⟶ a') :
+    (η.naturality (𝟙 a)).hom ▷ f ≫ (α_ _ _ _).hom ≫ η.app a ◁ G.mapId a ▷ f =
+    F.mapId a ▷ η.app a ▷ f ≫ (λ_ (η.app a)).hom ▷ f ≫ (ρ_ (η.app a)).inv ▷ f ≫
+    (α_ _ _ _).hom := by
+  apply η.toOplax.whiskerRight_naturality_id
+
+end
+
+/-- Vertical composition of strong natural transformations. -/
+@[simps!]
+def vcomp (η : StrongOplaxNatTrans F G) (θ : StrongOplaxNatTrans G H) : StrongOplaxNatTrans F H :=
+  mkOfOplax (OplaxNatTrans.vcomp η.toOplax θ.toOplax)
+    { naturality := λ {a b} f ↦
+        (α_ _ _ _).symm ≪≫ whiskerRightIso (η.naturality f) (θ.app b) ≪≫
+        (α_ _ _ _) ≪≫ whiskerLeftIso (η.app a) (θ.naturality f) ≪≫ (α_ _ _ _).symm }
+
+variable (B C)
+
+@[simps id comp]
+instance : CategoryStruct (Pseudofunctor B C) where
```

Comment:
As you are generating `simps` lemma, it may be appropriate to give an explicit name to this instance (and put it in the `Pseudofunctor` namespace), so that it will look better when we need to invoke these lemmas explicitly.

Reply:
Thanks for the suggestion, done!